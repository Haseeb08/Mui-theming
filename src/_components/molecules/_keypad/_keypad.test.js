import React from 'react';
import { create } from "react-test-renderer";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Keypad from "./_keypad";

describe("Keypad testing", () => {
  test("Matches the snapshot", () => {
    const input = create(<Keypad />);
    expect(input.toJSON()).toMatchSnapshot();
  });
  test("should have keypad", () => {
    const { getByTestId } = render(<Keypad />);
    expect(getByTestId("keypad")).toBeInTheDocument();
  });
  
  test("should have all buttons buttons", () => {
    const { getAllByTestId } = render(<Keypad />);
    expect(getAllByTestId("button")).toHaveLength(18);
  });

});