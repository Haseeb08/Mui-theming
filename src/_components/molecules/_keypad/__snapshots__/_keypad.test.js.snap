// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad testing Matches the snapshot 1`] = `
<div
  className="makeStyles-keyPad-1"
  data-testid="keypad"
>
  <div
    className="makeStyles-keyRow-2"
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-largeButton-4 [object Object]"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            AC
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-largeButton-4 [object Object]"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            &lt;=
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-keyRow-2"
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            9
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            8
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            7
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedSecondary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            +
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-keyRow-2"
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            6
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            5
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            4
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedSecondary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            -
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-keyRow-2"
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            3
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            2
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            1
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedSecondary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            *
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-keyRow-2"
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            0
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedPrimary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            .
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 makeStyles-equalButton-5"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            =
          </span>
        </span>
      </button>
    </div>
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-3 [object Object] MuiButton-containedSecondary"
        data-testid="button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiTypography-root MuiTypography-button"
          >
             
            /
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
